pydantic/main.py

```diff
@@ -658,6 +658,40 @@ def __eq__(self, other: Any) -> bool:
         else:
             return self.model_dump() == other
 
+    def model_copy(self, deep: bool = False) -> typing_extensions.Self:
+        """
+        Returns a copy of the model
+        """
+        return self.__deepcopy__() if deep else self.__copy__()
+
+    def __copy__(self) -> typing_extensions.Self:
+        """
+        Returns a shallow copy of the model
+        """
+        cls = type(self)
+        m = cls.__new__(cls)
+        _object_setattr(m, '__dict__', copy(self.__dict__))
+        _object_setattr(m, '__fields_set__', copy(self.__fields_set__))
+        for name in self.__private_attributes__:
+            value = getattr(self, name, Undefined)
+            if value is not Undefined:
+                _object_setattr(m, name, value)
+        return m
+
+    def __deepcopy__(self, memo: dict[int, Any] | None = None) -> typing_extensions.Self:
+        """
+        Returns a deep copy of the model
+        """
+        cls = type(self)
+        m = cls.__new__(cls)
+        _object_setattr(m, '__dict__', deepcopy(self.__dict__, memo=memo))
+        _object_setattr(m, '__fields_set__', copy(self.__fields_set__))
```

Comment:
this doesn't need to be a deepcopy because `__fields_set__` is a `set[str]`. I could use `deepcopy` for symmetry, but I figure that might be less performant in principle because python will probably attempt to deepcopy all the items from inside `__fields_set__` when we know that shouldn't be necessary.

I can add a comment explaining this if preferred.

Reply:
Added the comment; I'll mark this as resolved